tasks/test.js

```diff
@@ -0,0 +1,45 @@
+module.exports = function(grunt) {
+
+  var Drupal = require('../lib/drupal')(grunt);
+  var Help = require('../lib/help')(grunt);
+
+  // Default to Drupal 8 unless Drupal 7 is explicitly defined.
+  var script = Drupal.majorVersion() == 7 ? './scripts/run-tests.sh' : './core/scripts/run-tests.sh';
+  var testLocation = Drupal.majorVersion() == 7 ? './sites/all/modules/custom' : './modules/custom';
+  var domain = process.env.GDT_DOMAIN || grunt.config('config.domain') || require('os').hostname();
+
+  grunt.config('test', {
+    behat: true,
+    drupal: {
+      command: []
+    }
+  });
+
+  grunt.config(['shell', 'testdrupal'], {
+    command: grunt.config.get('config.testing.drupal.command') || 'php ' + script + ' --verbose --color --concurrency 4 --directory ' + testLocation + ' --url ' + domain,
+    options: {
+      execOptions: {
+        cwd: '<%= config.buildPaths.html %>'
+      }
+    }
+  });
+
+  // @todo Would rather display as 'alias for test:behat, test:drupal, etc'.
+  grunt.registerMultiTask('test', 'Run all configured tests.', function(args) {
```

Comment:
Run with `grunt test:behat`, the args value will be set to `behat`. This might be slightly more clean than the current `self.target`.